tests/ha/single_pod_restart/test_data_pod_restart_api.py

```diff
@@ -1221,3 +1222,249 @@ def test_obj_ver_suspension_data_pod_restart(self):
+                                          f"cluster status is not good")
+        LOGGER.info("Step 4: Successfully restarted the pod with replica method and checked "
+                    "cluster status")
+        self.restore_pod = False
+
+        LOGGER.info("Step 5: Overwrite existing object %s of bucket %s.", self.object_name,
+                    self.bucket_name)
+        if CMN_CFG["dtm0_disabled"]:
+            LOGGER.info("Step 5.1: Create new bucket and overwrite object in new bucket")
+            t_t = int(perf_counter_ns())
+            bucket_name = f"bucket-{t_t}"
+            object_name = f"object-{t_t}"
+            s3_data.update({bucket_name: [object_name, file_size]})
+        resp = self.ha_api.object_overwrite_dnld(self.s3_test_obj, s3_data, iteration=1,
+                                                 random_size=False)
+        assert_utils.assert_true(resp[0], "Failure observed in overwrite method.")
+        for checksum in resp[1].values():
+            assert_utils.assert_equal(checksum[0], checksum[1],
+                                      f"Checksum doesn't match, Expected: {checksum[0]} "
+                                      f"Received: {checksum[1]}")
+        LOGGER.info("Step 5: Successfully performed overwrite object operation after %s "
+                    "restart.", pod_name)
+
+        LOGGER.info("ENDED: Test to verify object overwrite before and after data pod restart")
+
+    @pytest.mark.ha
+    @pytest.mark.lc
+    @pytest.mark.tags("TEST-45638")
+    def test_obj_overwrite_during_data_pod_restart(self):
+        """
+        Verify object overwrite during data pod restart
+        """
+        LOGGER.info("STARTED: Test to verify object overwrite during data pod restart")
+        file_size = HA_CFG["5gb_mpu_data"]["file_size"]
+        output = Queue()
+        event = threading.Event()
+
+        LOGGER.info("Step 1: Create bucket %s and perform upload of object size %s MB and "
+                    "Overwrite the object", self.bucket_name, file_size)
+        s3_data = {self.bucket_name: [self.object_name, file_size]}
+        resp = self.ha_api.object_overwrite_dnld(self.s3_test_obj, s3_data, iteration=1,
+                                                 random_size=False)
+        assert_utils.assert_true(resp[0], "Failure observed in overwrite method.")
+        upld_chcksm = list(resp[1].values())[0][0]
+        for checksum in resp[1].values():
+            assert_utils.assert_equal(checksum[0], checksum[1],
+                                      f"Checksum doesn't match, Expected: {checksum[0]} "
+                                      f"Received: {checksum[1]}")
+        LOGGER.info("Step 1: Create bucket %s and perform upload of object size %s MB and "
+                    "Overwrite the object", self.bucket_name, file_size)
+
+        bkt_cnt = HA_CFG["copy_obj_data"]["bkt_multi"]
+        before_bkt = dict()
+        if CMN_CFG["dtm0_disabled"]:
```

Comment:
this is still healthy cluster so we don't need DTM flag here

Reply:
we only keep bucket created if dtm is there, otherwise we are creating new buckets in deg mode and performing over-write. That's why this conditon is check is there.
`new_s3_data = dict()
        if CMN_CFG["dtm0_disabled"]:
            LOGGER.info("Step 4: Start overwrite operation on new buckets")
            t_t = int(perf_counter_ns())
            for cnt in range(bkt_cnt):
                new_s3_data.update({f"ha-bkt{cnt}-{t_t}": [f"ha-obj{cnt}-{t_t}", file_size]})
        else:
            LOGGER.info("Step 4: Start overwrite operation on buckets created in healthy cluster")
            new_s3_data = before_bkt.copy()`